### PR TITLE
Centralized Media Management

### PR DESCRIPTION
**Pull request by francinum**
## About The Pull Request

Refactors configurable media track management into a centralized, shared system.

Also kills the disco jukebox, that thing's a dumpster fire.

## Why It's Good For The Game

Centralized management allows for tracks to be parsed exactly once, instead of repeatedly for the various purposes, as well as allowing sounds to be cleanly shared between pools, such as allowing various title tracks to be played in the IC jukeboxes.

## Changelog

:cl:
config: Credits music, Lobby tracks, and Jukebox media organization has changed massively. Check the repo for more information.
del: Disco Inferno has been made far less disco.
/:cl:

### CODE REVIEW

**Comment by gonenoculer5** [comment]
I can't believe they killed disco inferno

File: code/controllers/subsystem/ticker.dm [hunk 1]
```diff
@@ -683,103 +683,44 @@ SUBSYSTEM_DEF(ticker)
...
-
-	//Remove the old login music from the current pool if it wouldn't empty the pool.
+	var/list/title_music_data = SSmedia.get_track_pool(MEDIA_TAG_LOBBYMUSIC_COMMON)
+	var/list/rare_music_data = SSmedia.get_track_pool(MEDIA_TAG_LOBBYMUSIC_RARE)
+	///The full path of the last song used
+	var/old_login_music_t
+	///The full datum of the last song used.
+	var/datum/media/old_login_music
+
+	if(fexists("data/last_round_lobby_music.txt"))
```
Comment (Kapu1178) [inline code/controllers/subsystem/ticker.dm:707]:
```suggestion
	if(rustg_file_exists("data/last_round_lobby_music.txt"))
```

File: code/controllers/subsystem/ticker.dm [hunk 2]
```diff
@@ -683,103 +683,44 @@ SUBSYSTEM_DEF(ticker)
...
+		old_login_music_t = file2text("data/last_round_lobby_music.txt")
```
Comment (Kapu1178) [inline code/controllers/subsystem/ticker.dm:708]:
```suggestion
		old_login_music_t = rustg_file_read("data/last_round_lobby_music.txt")
```

File: code/controllers/subsystem/media.dm [hunk 3]
```diff
@@ -0,0 +1,152 @@
...
+
+/datum/controller/subsystem/media/Initialize(start_timeofday)
+	//I'm not even going to bother supporting the existing jukebox shit. Jsons are easier.
+	tracks_by_tag = list()
+	var/basedir = "[global.config.directory]/media/jsons/"
+	var/invalid_jsons_exist = FALSE
+	//Fetch
+	for(var/json_record in flist(basedir))
+		//Decode
+		var/list/json_data = json_decode(file2text("[basedir][json_record]"))
```
Comment (Kapu1178) [inline code/controllers/subsystem/media.dm:39]:
```suggestion
		var/list/json_data = json_decode(rustg_file_read("[basedir][json_record]"))
```

File: code/controllers/subsystem/media.dm [hunk 4]
```diff
@@ -0,0 +1,152 @@
...
+		/// A two-entry list containing the erroring tag, and a reason for the error.
+		var/tag_error
+		for(var/jd_tag in jd_tag_cache)
+			switch(jd_tag)
+
+				//Validation relevant for ALL tracks.
+				if(MEDIA_TAG_ALLMEDIA)
+					if(!json_data["name"])
+						tag_error = list(MEDIA_TAG_ALLMEDIA, "Track has no name.")
+					if(!fexists(jd_full_filepath))
```
Comment (Kapu1178) [inline code/controllers/subsystem/media.dm:59]:
```suggestion
					if(!rustg_file_exists(jd_full_filepath))
```

**Review by Kapu1178 — CHANGES_REQUESTED** [review]
rustg

**Pull request merged**